src/goto-symex/goto_symex.h

```diff
@@ -63,22 +86,27 @@ class goto_symext
   {
   }
 
-  typedef goto_symex_statet statet;
-
   /** symex all at once, starting from entry point */
-  virtual void operator()(
-    const goto_functionst &goto_functions);
+  virtual void symex_from_entry_point_of(
+    const goto_functionst &goto_functions,
+    symbol_tablet &new_symbol_table);
 
-  /** symex starting from given goto program */
-  virtual void operator()(
+  /// Performs symbolic execution using a state and equation that have
+  /// already been used to symex part of the program. The state is not
+  /// re-initialized; instead, symbolic execution resumes from the program
+  /// counter of the saved state.
+  virtual void resume_symex_from_saved_state(
     const goto_functionst &goto_functions,
-    const goto_programt &goto_program);
+    const statet &saved_state,
+    symex_target_equationt *const saved_equation,
+    symbol_tablet &new_symbol_table);
 
   /** start symex in a given state */
```

Comment:
Good opportunity to improve this comment? For example, if we start from here, where do we continue until?